test/e2e/util.go

```diff
@@ -1137,6 +1157,20 @@ func RunRC(config RCConfig) error {
 
 		Logf("%v %v Pods: %d out of %d created, %d running, %d pending, %d waiting, %d inactive, %d unknown ",
 			time.Now(), rc.Name, len(pods), config.Replicas, running, pending, waiting, inactive, unknown)
+
+		runningMetric.Set(float64(running))
+		pendingMetric.Set(float64(pending))
+
+		if testContext.PrometheusPushGateway != "" {
```

Comment:
Is there a way to make this work without sprinkling these if blocks all over our code base? This is going to be hard to read; also we definitely will need some tests if you want this to keep working.